### PR TITLE
Add rematch button at the end of a game

### PR DESCRIPTION
**Pull request by bglw**

### CODE REVIEW

File: truncate_core/src/board.rs [hunk 1]
```diff
@@ -270,6 +270,20 @@ impl Board {
         }
     }
 
+    pub fn get_mut<'a>(
```
Comment (BlakeMScurr) [inline truncate_core/src/board.rs:273]: Confused how this didn't already exist, but all g. I guess we used a `set`ter?

File: truncate_server/src/main.rs [hunk 2]
```diff
@@ -321,6 +321,74 @@ async fn handle_player_msg(
...
+                                player_index: i,
+                                room_code: new_game_id.clone(),
+                            },
+                            Duration::from_days(7), // TODO: Determine game expiration time
+                        );
+                        let token = jwt_key
+                            .authenticate(claims)
+                            .expect("Claims should be serializable");
+
+                        peer.send(GameMessage::JoinedLobby(
```
Comment (BlakeMScurr) [inline truncate_server/src/main.rs:380]: I guess if there's a broken connection when we're rematching you'd never get your jwt, and you could never join the new lobby, right? But not a big issue - just remake.

I'm just wondering why the old jwt can't be used for the new lobby?

**Review by BlakeMScurr — APPROVED** [review]
LGTM

File: truncate_core/src/board.rs (hunk 1 above)
Comment (bglw) [reply truncate_core/src/board.rs:273]: Yeah we always `set`, or I think reach into the Vecs by hand to modify

File: truncate_server/src/main.rs (hunk 2 above)
Comment (bglw) [reply truncate_server/src/main.rs:380]: The JWT claims stores the room code and the player index, so we need to make a new one against the new details. (We don't store the JWTs, so we rely on the information within to reconnect to the correct game as the correct player)

Yeah some issue there with rematching not applying, but for now thems the breaks

**Pull request merged**